providers/SupportedWalletProvider.tsx

```diff
@@ -0,0 +1,261 @@
+import { hexToString } from "@polkadot/util";
+import ERC20Tokens from "../artifacts/erc20tokens.json";
+
+export type BalanceInfo = AssetInfo & {
+  value: number;
+};
+
+type WalletContext = {
+  balances: Array<BalanceInfo>;
+  selectedAccount: InjectedAccountWithMeta;
+  wallet: InjectedExtension;
+  connectWallet: (callback?: () => void) => Promise<void>;
+  disconnectWallet: () => void;
+  selectAccount: (account: InjectedAccountWithMeta) => void;
+  setBalances: Function;
+  bridgeBalances: Object;
+  signer: Signer;
+};
+
+const SupportedWalletContext = createContext<WalletContext>({
+  balances: null,
+  selectedAccount: null,
+  wallet: null,
+  connectWallet: null,
+  disconnectWallet: null,
+  selectAccount: null,
+  setBalances: null,
+  bridgeBalances: null,
+  signer: null,
+});
+
+type ProviderProps = {};
+
+export default function SupportedWalletProvider({
+  children,
+}: PropsWithChildren<ProviderProps>) {
+  const { browser } = useUserAgent();
+  const { api } = useCENNZApi();
+  const accounts = useWeb3Accounts();
+  const { web3Enable, web3FromSource } = useDappModule();
+  const [signer, setSigner] = useState<Signer>(null);
+  const [wallet, setWallet] = useState<InjectedExtension>(null);
+  const [selectedAccount, setAccount] = useState<InjectedAccountWithMeta>(null);
+
+  const connectWallet = useCallback(
+    async (callback) => {
+      if (!web3Enable || !api) return;
+
+      const extension = (await web3Enable("Litho")).find(
+        (extension) => extension.name === "cennznet-extension"
+      );
+
+      const injector = await web3FromSource("cennznet-extension");
+      setSigner(injector.signer);
+
+      if (!extension) {
+        const confirmed = confirm(
+          "Please install the CENNZnet extension then refresh the page."
+        );
+
+        if (!confirmed) return;
+
+        window.open(
+          browser.name === "Firefox"
+            ? "https://addons.mozilla.org/en-US/firefox/addon/cennznet-browser-extension/"
+            : "https://chrome.google.com/webstore/detail/cennznet-extension/feckpephlmdcjnpoclagmaogngeffafk",
+          "_blank"
+        );
+
+        return;
+      }
+
+      if (callback) callback();
+      setWallet(extension);
+      store.set("CENNZNET-EXTENSION", extension);
+    },
+    [web3Enable, browser, api]
+  );
+
+  const disconnectWallet = useCallback(() => {
+    if (!confirm("Are you sure?")) return;
+    store.remove("CENNZNET-EXTENSION");
+    store.remove("CENNZNET-ACCOUNT");
+    store.remove("EXTENSION_META_UPDATED");
+    setWallet(null);
+    setAccount(null);
+    setBalances(null);
+  }, []);
+
+  const selectAccount = useCallback((account) => {
+    setAccount(account);
+    store.set("CENNZNET-ACCOUNT", account);
+  }, []);
+
+  // 1. Restore the wallet from the store if it exists
+  useEffect(() => {
+    const storedWallet = store.get("CENNZNET-EXTENSION");
+    if (storedWallet) setWallet(storedWallet);
```

Comment:
I've made some changes here https://github.com/cennznet/litho/pull/141/commits/66e3999791a5c3dc709d2f5c8b7fe6d4fcbae231 so when restoring the wallet, we also restore all of its functionality (e.g `waller.signer`), can you update it?

Reply:
done